smartctl.go

```diff
@@ -42,6 +43,16 @@ type SMARTctl struct {
 	device SMARTDevice
 }
 
+func extractDiskName(input string) string {
+	re := regexp.MustCompile(`^(/dev/\S+/\S+\s\[|/dev/)(?:\s\[|)(?P<disk>\S+)(?:\]\s.*|)$`)
```

Comment:
good start, but let's try to come up with a clear plan here, esp for drives where --scan doesn't work per my previous output.

If we can find a way to test the case of multiple megaraid controllers, and see if the IDs are unique, that would be really informative. 

The disk label should uniquely identify a drive in a system for a given boot, until the drive is re-detected, or the host reboots (both of which might number it differently).

The disk label should be used by Prometheus consumers to join with the drive information e.g. serial number, or other metrics like node-exporter iostat data.

Reply:
>good start, but let's try to come up with a clear plan here, esp for drives where --scan doesn't work per my previous output.

Suggest that could be a separate issue/PR.

>If we can find a way to test the case of multiple megaraid controllers, and see if the IDs are unique, that would be really informative.

If there are multiple HBAs, the IDs on each likely will start with 0.  Do you mean unique across HBAs?  I doubt that will happen.   One also sees multiple enclosures under a given HBA, but I suspect that doesn't matter to us here.

>The disk label should be used by Prometheus consumers to join with the drive information e.g. serial number, or other metrics like node-exporter iostat data.

I think you're a much bigger fan of `group_left` than I am.

Reply:
How long time to wait merge?